### PR TITLE
Change http status codes to object that with readable keys

### PR DESCRIPTION
**Pull request by rallu**
No need to remember http error codes! And even easier with IDE auto complete tools.

### CODE REVIEW

**Comment by cognitom** [comment]
Hi Juha!
'404' looks easy for me 😆 
Beside my preference, I agree it's good to make it more readable.

How about just adding code like this? I like to keep things small.

```javascript
export const code = Object.keys(httpStatusCodes).reduce((acc, key) => {
  const words = httpStatusCodes[key].split(' ')
  const camel = words.shift().toLowerCase() + words.join('')
  acc[camel] = key
  return acc
}, {})
```

**Comment by cognitom** [comment]
Expected usages:

```javascript
import HttpError, {code} from 'es-http-error'

try { throw new HttpError(code.notFound) }
catch (e) {
  if (e.code = code.notFound) goHome()
}
```

File: README.md [hunk 1]
```diff
@@ -35,13 +36,20 @@ async function main (id) {
     const render(view(await res.json()), document.body)
   } catch (e) {
     if (e instanceof HttpError) {
-      if (e.code === 401) logout()
-      if (e.code === 404) notFound()
+      if (e.type === HttpStatusCodes.unauthorized) logout()
+      if (e.type === HttpStatusCodes.not-found) notFound()
```
Comment (cognitom) [inline README.md]: We can't use `param-case` here. `camelCase` seems better.

**Comment by rallu** [comment]
That function looks like what I did for the source data to create that object 😃. I just wonder that if that solutions still makes IDE's to do the auto complete?

File: README.md (hunk 1 above)
Comment (rallu) [reply README.md]: I agree.

**Comment by cognitom** [comment]
Oh, that's a good catch. OK, then, how about this?:

```javascript
/** Name to code */
export const code = {
  unauthorized: 401,
  notFound: 404,
  ...
}
/** Code to English */
const dictionary = {
  401: 'Unauthorized',
  404: 'Not Found',
  ...
}
export default class HttpError extends Error {...}
```

Pros:
- Easy to extend other languages
- Keeps small size with treeshaking

**Comment by rallu** [comment]
That looks very cool way of doing it! Do you or I do it? :)

**Comment by cognitom** [comment]
Thanks. Pls!
But, no rush :-)
I'll go to bed now. It's 1am Tokyo. See ya...zzz

**Comment by cognitom** [comment]
Thanks! I've done merge and added some features as well.
Before `npm publish`, I'm rethinking `httpCode` is the best term or not. `statusCode` feels slightly natural for me. What do you think?

**Comment by rallu** [comment]
Yes. `statusCode` sounds better to me too.

**Comment by cognitom** [comment]
OK, done. f8dcabda57980e3d2d36fbefdaf6e2f34f3c734a
I'll publich later. Anyway thanks for your feedbacks!

**Pull request merged**